### PR TITLE
fix: MISRA 15.1 (goto) violations - part 1

### PR DESCRIPTION
**Pull request by abhidixi11**
Removes goto from some files.
This PR also contains a unit test fix. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c [hunk 1]
```diff
@@ -2289,7 +2289,11 @@ TEST( MQTT_Unit_API, GetIncomingMQTTPacketTypeAndLengthChecks )
 
     /* Test with invalid remaining length. */
     buffer[ 0 ] = 0x20; /* CONN ACK */
-    buffer[ 1 ] = 0xFF; /* Remaining length. */
+    /* To generate invalid remaining length response,
+     * three bytes need to have MSB (or continuation bit, 0x80) set */
+    buffer[ 1 ] = 0xFF;
+    buffer[ 2 ] = 0xFF;
```
Comment (gkwicker) [inline libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c]: This change also seems unrelated to removing `goto`, so please call it out in the commit message.

Comment (abhidixi11) [reply libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c]: It is in the commit message , "PR also contains a unit test fix."

File: libraries/standard/mqtt/src/iot_mqtt_network.c [hunk 2]
```diff
@@ -256,25 +256,24 @@ static IotMqttError_t _getIncomingPacket( void * pNetworkConnection,
...
-    if( pIncomingPacket->remainingLength == MQTT_REMAINING_LENGTH_INVALID )
+    if( status == IOT_MQTT_SUCCESS )
     {
-        status = IOT_MQTT_BAD_RESPONSE;
-        goto cleanup;
+        /* Read the remaining length. */
+        pIncomingPacket->remainingLength = _getRemainingLengthFunc( pMqttConnection->pSerializer )( pNetworkConnection,
+                                                                                                    pMqttConnection->pNetworkInterface );
+
+        if( pIncomingPacket->remainingLength == MQTT_REMAINING_LENGTH_INVALID )
```
[the remark this answers is not in the document]
Comment (abhidixi11) [reply libraries/standard/mqtt/src/iot_mqtt_network.c]: It is only in one place inside _getIncomingPacket().

File: libraries/standard/mqtt/src/iot_mqtt_network.c [hunk 3]
```diff
@@ -286,27 +285,37 @@ static IotMqttError_t _getIncomingPacket( void * pNetworkConnection,
                          pMqttConnection,
                          ( unsigned long ) pIncomingPacket->remainingLength,
                          ( unsigned long ) pIncomingPacket->type );
```
Comment (leegeth) [inline libraries/standard/mqtt/src/iot_mqtt_network.c]: This log seems to have 2 copies and the above doesn't seems at the right place. You may want to add the coverity annotations to the right one below

File: libraries/standard/mqtt/src/iot_mqtt_network.c [hunk 4]
```diff
@@ -286,27 +285,37 @@ static IotMqttError_t _getIncomingPacket( void * pNetworkConnection,
...
+            pIncomingPacket->pRemainingData = IotMqtt_MallocMessage( pIncomingPacket->remainingLength );
 
-            _flushPacket( pNetworkConnection, pMqttConnection, pIncomingPacket->remainingLength );
+            if( pIncomingPacket->pRemainingData == NULL )
+            {
+                IotLogError( "(MQTT connection %p) Failed to allocate buffer of length "
+                             "%lu for incoming packet type %lu.",
+                             pMqttConnection,
+                             ( unsigned long ) pIncomingPacket->remainingLength,
+                             ( unsigned long ) pIncomingPacket->type );
```
Comment (leegeth) [inline libraries/standard/mqtt/src/iot_mqtt_network.c]: please see the comment above

Comment (abhidixi11) [reply libraries/standard/mqtt/src/iot_mqtt_network.c]: Yes, that's the auto merge issue from earlier PR.

Comment (leegeth) [reply libraries/standard/mqtt/src/iot_mqtt_network.c]: Yea. Thought so
